### PR TITLE
📖 Extend packaging.md

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR tries to fill out the description of the OCM status addon, and discovers that there is not much ready to say about the OCM transport plugin.

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by ezrasilvera** [comment]
@MikeSpreitzer This is a very detailed explanation of the package relations. I assume we will also have a simpler guide just explaining how to publish the images and what to be aware of when creating when creating images or releases - taking into account all the dependencies mentioned in this doc.

**Comment by MikeSpreitzer** [comment]
@ezrasilvera : I am not sure about a doc focused on just how to build images. Frankly, I have open questions about how images should be built. I expect there will be a doc about how to make a release. I am not sure how interconnected things, and thus doc about them, will be. Do not yet want to promise that building an image is a meaningful thing to doc on its own.

**Comment by MikeSpreitzer** [comment]
More generally about other documentation: we have open design questions to settle about our operations, and documentation to write about that after or as part of resolving those questions. All that needs to be informed by the information about the packaging and dependencies and the existing procedures for building and publishing. I am starting with documenting that.

**Comment by MikeSpreitzer** [comment]
The force-push to 49a9b404f updates the description of the versions in the ocm-status-addon-chart.

**Comment by MikeSpreitzer** [comment]
The force-push to bb63972d0 recognizes that helm enforces the equation that the OCI tag of a chart image equals the "version" in Chart.yaml.

File: docs/content/v0.20/packaging.md [hunk 1]
```diff
@@ -9,48 +9,135 @@ Note that to _use_ KubeStellar, all of the repos in this graph are involved.
...
 
-.. to be written ...
+**NOTE**: This is under development, not a part of the functioning
+  system yet.
+
+The source is the GitHub repo [github.com/kubestellar/ocm-transport-plugin](https://github.com/kubestellar/ocm-transport-plugin)
+
+### OCM Transport Plugin container image
+
+This will apsirationally appear at [ghcr.io/kubestellar/ocm-transport-plugin](https://github.com/orgs/kubestellar/packages/container/package/ocm-transport-plugin). This level of development has not been reached yet.
```
Comment (pdettori) [inline docs/content/v0.20/packaging.md:104]: typo: apsirationally -> aspirationally

File: docs/content/v0.20/packaging.md [hunk 2]
```diff
@@ -9,48 +9,135 @@ Note that to _use_ KubeStellar, all of the repos in this graph are involved.
...
+### OCM status addon container image
+
+There is a container image at [ghcr.io/kubestellar/ocm-status-addon](https://github.com/orgs/kubestellar/packages/container/package/ocm-status-addon).
+
+The container image is built and published using `make`.  The
+following `make` variables are relevant.
+
+- `KO_DOCKER_REPO`, defaults to `ghcr.io/kubestellar`
+- `CMD_NAME`, defaults to `ocm-status-addon`
+- `IMAGE_TAG`, defaults to `0.2.0-alpha.1`
```
Comment (pdettori) [inline docs/content/v0.20/packaging.md:51]: This reminds me that that need to be changed in the next build to `v0.20.0-alpha.1`

File: docs/content/v0.20/packaging.md [hunk 3]
```diff
@@ -9,48 +9,135 @@ Note that to _use_ KubeStellar, all of the repos in this graph are involved.
...
+- `IMG`, defaults to `${KO_DOCKER_REPO}/${CMD_NAME}:${IMAGE_TAG}`
+
+`make ko-build-push` will build and publish the container image at
+`${IMG}`. This will be a multi-platform manifest, referencing images
+for "platforms" `linux/amd64` and `linux/arm64`. This will not work if
+you have `podman` aliased to `docker`.
```
Comment (pdettori) [inline docs/content/v0.20/packaging.md:57]: AFAIK ko build and push to a repo does not depend on docker. It depends on docker if you do a local build that stores the image on the local docker registry.

File: docs/content/v0.20/packaging.md [hunk 4]
```diff
@@ -9,48 +9,135 @@ Note that to _use_ KubeStellar, all of the repos in this graph are involved.
...
+
+`make ko-local-build` will build the same multi-platform manifest but
+not push it, only leave it among your Docker images. This also will
+not work if you have `podman` aliased to `docker`.
+
```
Comment (pdettori) [inline docs/content/v0.20/packaging.md:62]: ko local build requires communication with the docker unix socket, so this is it is needed to specify the socket address.

File: docs/content/v0.20/packaging.md [hunk 5]
```diff
@@ -9,48 +9,135 @@ Note that to _use_ KubeStellar, all of the repos in this graph are involved.
...
+The operator is delivered by a Helm chart at [ghcr.io/kubestellar/ocm-status-addon-chart](https://github.com/orgs/kubestellar/packages/container/package/ocm-status-addon-chart).
 
-The [status-addon](https://github.ibm.com/dettori/status-addon) repo is the source of a RedHat-style operator.
+The chart is built and published using `make`. The following `make`
+variables are relevant.
 
-The operator is delivered by a Helm chart at `quay.io/pdettori/status-addon-chart`. Chart versioning? How is the chart built, pushed?
+- `KO_DOCKER_REPO`, defaults to `ghcr.io/kubestellar`
+- `CMD_NAME`, defaults to `ocm-status-addon`
+- `IMAGE_TAG`, defaults to `0.2.0-alpha.1`
```
Comment (pdettori) [inline docs/content/v0.20/packaging.md:73]: same comments on the tag version as above, need to republish with v0.20.0-alpha.1

File: docs/content/v0.20/packaging.md [hunk 6]
```diff
@@ -9,48 +9,135 @@ Note that to _use_ KubeStellar, all of the repos in this graph are involved.
...
+- `IMG`, defaults to `${KO_DOCKER_REPO}/${CMD_NAME}:${IMAGE_TAG}`
 
-There is also a container image involved.
+`make chart`: builds the local copy of the chart from local sources,
+using `kustomize`. When using the default values of the `make`
+variables, the built chart references the container image
+`ghcr.io/kubestellar/ocm-status-addon:0.2.0-alpha.1`. TODO: describe
+how the chart contents and destination depend on the `make` variables.
+
```
Comment (pdettori) [inline docs/content/v0.20/packaging.md:81]: The image reference is updated when `make chart` is run by kustomize using the IMG variable

File: docs/content/v0.20/packaging.md [hunk 7]
```diff
@@ -9,48 +9,135 @@ Note that to _use_ KubeStellar, all of the repos in this graph are involved.
...
+[ghcr.io/kubestellar/ocm-status-addon-chart](https://github.com/orgs/kubestellar/packages/container/package/ocm-status-addon-chart). How
+is the OCI tag of the chart controlled?
+
+The following versions exist.
+
+| OCI tag = version in chart | appVersion in chart | referenced container tag |
+| -------------------------- | ------------------- | ------------------------ |
+| 0.2.0-alpha.1              | 0.2.0-alpha.1       | 0.2.0-alpha.1 |
+
+Note: the Chart.yaml in github uses just "0.2.0" for the version and appVersion in the chart. TODO: bug?
```
Comment (pdettori) [inline docs/content/v0.20/packaging.md:93]: indeed

**Comment by francostellari** [comment]
I think we need to decide/standardize the container images to have or not have the `v` in front of the version.
So do, some don't.
For example the text mentions a `helm:v3.14.0` but we have a `helm:3.14.0`

**Comment by clubanderson** [comment]
my preference is to leave out the v.

**Comment by MikeSpreitzer** [comment]
I vaguely recall that something relevant insists on the "v" prefix, but do not recall what that is right now.

File: docs/content/v0.20/packaging.md (hunk 2 above)
Comment (MikeSpreitzer) [reply docs/content/v0.20/packaging.md:51]: I do not think we have a requirement of equality between release identifiers in different repos. In fact I think the opposite: we should definitely have the freedom to make a new release of a repo X with also making a new release of every repo that X depends on.

File: docs/content/v0.20/packaging.md (hunk 5 above)
Comment (MikeSpreitzer) [reply docs/content/v0.20/packaging.md:73]: Same responses.

File: docs/content/v0.20/packaging.md (hunk 7 above)
Comment (MikeSpreitzer) [reply docs/content/v0.20/packaging.md:93]: I think this bug is item (4) in https://github.com/kubestellar/ocm-status-addon/issues/11

**Comment by MikeSpreitzer** [comment]
The force-push to a8aa77211 makes changes responding to the review above.

File: docs/content/v0.20/packaging.md (hunk 1 above)
Comment (MikeSpreitzer) [reply docs/content/v0.20/packaging.md]: fixed, thanks.

File: docs/content/v0.20/packaging.md (hunk 3 above)
Comment (MikeSpreitzer) [reply docs/content/v0.20/packaging.md]: Fixed.

File: docs/content/v0.20/packaging.md (hunk 4 above)
Comment (MikeSpreitzer) [reply docs/content/v0.20/packaging.md:62]: Revised, trying to more clearly say what I meant.

File: docs/content/v0.20/packaging.md (hunk 6 above)
Comment (MikeSpreitzer) [reply docs/content/v0.20/packaging.md:81]: That will be easier to follow once https://github.com/kubestellar/ocm-status-addon/issues/9 is fixed.

**Comment by MikeSpreitzer** [comment]
Regarding starting with "v", here's a biggie: https://go.dev/ref/mod#versions

File: docs/content/v0.20/packaging.md (hunk 2 above)
Comment (pdettori) [reply docs/content/v0.20/packaging.md:51]: good point, indeed, we should have independent versioning for each repo (as it is currently for kubeflex)

**Review by pdettori — APPROVED** [review]
/lgtm

**Comment by MikeSpreitzer** [comment]
@clubanderson , @francostellari : the question of what we _should_ do what not my intent to address here, this was only intended to document the present truth. Let's continue to discuss what we _should_ do under #1732 .

**Pull request merged**